### PR TITLE
Make integration tests run more stable

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
To make the CI more reliable.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
Fixes https://github.com/containers/conmon-rs/issues/342
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: conmon-rs/server/src/server.rs [hunk 1]
```diff
@@ -174,18 +174,16 @@ impl Server {
...
 
+        debug!("Killing grandchildren");
+        reaper
+            .kill_grandchildren(handled_sig)
+            .context("could not kill grandchildren")?;
+
+        debug!("Sending shutdown message");
         shutdown_tx
             .send(())
             .map_err(|_| format_err!("unable to send shutdown message"))?;
```
Comment (saschagrunert) [inline conmon-rs/server/src/server.rs:185]: I assume it was a race between those two.

File: conmon-rs/server/src/child_reaper.rs [hunk 2]
```diff
@@ -233,15 +236,18 @@ impl ReapableChild {
...
+        debug!("Waiting for tasks to close");
         if let Some(t) = self.task.clone() {
             let tasks = lock!(t).take().context("no tasks available")?;
             for t in tasks.into_iter() {
-                debug!("Grandchild await");
-                t.await?;
+                debug!("Task await");
+                if let Err(e) = t.await {
+                    warn!("Unable to wait for task: {:#}", e)
+                }
```
Comment (saschagrunert) [inline conmon-rs/server/src/child_reaper.rs:246]: I think we should try to finish all tasks even if one fails.

**Comment by saschagrunert** [comment]
Ready for review. PTAL @rphillips @haircommander

**Comment by haircommander** [comment]
one nit, otherwise LGTM, nice investigation here :)

File: conmon-rs/server/src/child_reaper.rs [hunk 3]
```diff
@@ -142,22 +142,25 @@ impl ChildReaper {
...
             futures::executor::block_on(
                 async {
                     if let Err(e) = grandchild.close().await {
-                        error!("Unable to close grandchild: {}", e)
+                        error!("Unable to close grandchild: {:#}", e)
                     }
                 }
                 .instrument(debug_span!("close", signal = s.as_str())),
             );
+            debug!("Done");
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply conmon-rs/server/src/child_reaper.rs]: Yep, done :+1:

**Pull request merged**